### PR TITLE
fix(gcs): throw argo not found error if key not exist

### PR DESCRIPTION
**Pull request by AntoineDao**
Checklist:

* [x] My organization is added to [USERS.md](https://github.com/argoproj/argo-workflows/blob/master/USERS.md).

re #6276 

I took the liberty of not adding tests because it's a 1 line change and there are currently no tests for the gcs artifact driver :stuck_out_tongue: 

@tymokvo I know you implemented this changed and asked whether we should submit a PR, I went to the liberty of just submitting it and letting the maintainers decide whether or not they want it merged :smile:

### CODE REVIEW

**Comment by alexec** [comment]
This looks ok to me? @sarabala1979

File: workflow/artifacts/gcs/gcs.go [hunk 1]
```diff
@@ -87,6 +87,10 @@ func downloadObjects(client *storage.Client, bucket, key, path string) error {
 	if err != nil {
 		return err
 	}
+	if len(objNames) < 1 {
```
[the remark this answers is not in the document]
Comment (tymokvo) [reply workflow/artifacts/gcs/gcs.go:90]: What do you mean by "it should just come out"? It looks like whether or not it is optional is handled by the executor in this [line](https://github.com/argoproj/argo-workflows/blob/2c2525981e046bf8afdef4e1f007a32e88722914/workflow/executor/executor.go#L184).

[the remark this answers is not in the document]
Comment (sarabala1979) [reply workflow/artifacts/gcs/gcs.go:90]: Got it I missed that place

**Review by sarabala1979 — APPROVED** [review]
LGTM

**Pull request merged**